R/system.R

```diff
@@ -672,57 +666,53 @@ getAmazonAthenaConnection <- function(driver = "", region = "", authenticationTy
   if(additionalParams != "") {
     connectionString <- stringr::str_c(connectionString, ";", additionalParams)
   }
-  if(timezone != "") {
-    connectionString <- stringr::str_c(connectionString, ";", timezone)
+  if (timezone == "") {
+    timezone <- "UTC" # if timezone is not provided use UTC as default timezone. This is also the default for odbc::dbConnect.
   }
+  connectionString <- stringr::str_c(connectionString, ";timezone=", timezone)
+  connectionString <- stringr::str_c(connectionString, ";timezone_out=", timezone)
+
   if (endpointOverride != "") {
-    connectionString <- stringr::str_c(connectionString, ";", endpointOverride)
+    connectionString <- stringr::str_c(connectionString, ";EndpointOverride=", endpointOverride)
   }
 
+  # For Windows, set encoding to make sure non-ascii data is handled properly.
+  # ref: https://github.com/r-dbi/odbc/issues/153
+  if (is.win <- Sys.info()['sysname'] == 'Windows') {
+    loc <- Sys.getlocale(category = "LC_CTYPE")
+    # loc looks like "Japanese_Japan.932", so split it with dot ".".
+    encoding <- stringr::str_split(loc, pattern = "\\.")
+    if (length(encoding[[1]]) == 2) {
+      connectionString <- stringr::str_c(connectionString, ";encoding=", encoding[[1]][[2]])
```

Comment:
Check if encoding in connection string works fine.

Reply:
Checked and encoding need to be passed as a separate parameter.